core/akka-runtime/src/main/scala/com/ing/baker/runtime/akka/AkkaBaker.scala

```diff
@@ -450,15 +450,15 @@ class AkkaBaker private[runtime](config: AkkaBakerConfig) extends scaladsl.Baker
    * @param implementation The implementation object
    */
   override def addInteractionInstance(implementation: InteractionInstance): Future[Unit] =
-    Future.successful(config.interactionManager.addImplementation(implementation))
+    config.interactionManager.addImplementation(implementation)
```

Comment:
Not strictly necessary, as the service discovery provides alternative implementation, but it reduces unnecessary concurrency.

Reply:
ah yhea, this was actually unintended, on Baker 3.0 we moved the api to an async based api (basically added Futures in a bunch of places) but the interaction manager lag behind that, its interface used not to be async but baker was already async, that is why there was a temporary Future.successful there. So good find hehe